File: GoLandProjects/algorithm-sort-go/algorithm/sort/util_test.go
package sort

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestLess(t *testing.T) {
	arr := []int{1, 2}
	assert.True(t, less(arr, 0, 1))
}

func TestExch(t *testing.T) {
	arr := []int{1, 2}
	exch(arr, 0, 1)
	assert.Equal(t, arr[0], 2)
	assert.Equal(t, arr[1], 1)
	assert.Equal(t, len(arr), 2)
}

File: tests/util-test.rs
#[test]
fn test_less() {
    let v = vec![1, 2];
    assert_eq!(algorithm_sort_rust::less(v, 0, 1), true);
}

#[test]
fn test_exch() {
    let mut v = vec![1, 2];
    algorithm_sort_rust::exch(&mut v, 0, 1);
    assert_eq!(v[0], 2);
    assert_eq!(v[1], 1);
    assert_eq!(v.len(), 2);
}
